app/src/main/java/com/mikhailovskii/weatherandroid/data/firebase/FirebaseModel.kt

```diff
@@ -0,0 +1,188 @@
+                    }.addOnFailureListener { e ->
+                        Timber.e("Error twitter save: $e")
+                    }
+            }
+
+            callback?.onLoggedInWithFilledInfo()
+        }.addOnFailureListener {
+            callback?.onLoginFailed()
+        }
+
+    }
+
+    fun logInWithFacebook(user: User, callback: FirebaseLoginCallback?) {
+        db.collection(USERS_COLLECTION).get()
+            .addOnSuccessListener { databaseResult ->
+                var isUserPresents = false
+
+                loop@ for (document in databaseResult) {
+                    val databaseUser = document.toObject(User::class.java)
+
+                    if (databaseUser.login == user.login
+                        && databaseUser.facebookKey == user.facebookKey
+                        && databaseUser.icon == user.icon
+                    ) {
+                        isUserPresents = true
+                        break@loop
+                    }
+                }
+
+                if (!isUserPresents) {
+                    db.collection(USERS_COLLECTION).document().set(user)
+                        .addOnSuccessListener {
+                            Timber.d(("Facebook info saved"))
+                        }.addOnFailureListener { e ->
+                            Timber.e("Error facebook save: $e")
+                        }
+                }
+
+                callback?.onLoggedInWithFilledInfo()
+            }.addOnFailureListener {
+                callback?.onLoginFailed()
+            }
+    }
+
+    fun logInWithGoogle(user: User, callback: FirebaseLoginCallback?) {
+        db.collection(USERS_COLLECTION).get().addOnSuccessListener { databaseResult ->
+            var isUserPresents = false
+
+            loop@ for (document in databaseResult) {
+                val databaseUser = document.toObject(User::class.java)
+
+                if (databaseUser.login == user.login
+                    && databaseUser.googleKey == user.googleKey
+                    && databaseUser.icon == user.icon
+                ) {
+                    isUserPresents = true
+                    break@loop
+                }
+            }
+
+            if (!isUserPresents) {
+                db.collection(USERS_COLLECTION).document().set(user)
+                    .addOnSuccessListener {
+                        Timber.d(("Facebook info saved"))
+                    }.addOnFailureListener { e ->
+                        Timber.e("Error facebook save: $e")
+                    }
+            }
+
+            callback?.onLoggedInWithFilledInfo()
+        }.addOnFailureListener {
+            callback?.onLoginFailed()
+        }
+    }
+
+    fun getStickerPackByName(name: String, callback: FirebaseDataCallback?) {
+        db.collection(STICKERS_COLLECTION)
+            .whereEqualTo(TITLE_FIELD, name)
+            .get()
+            .addOnSuccessListener { result ->
+                if (!result.isEmpty) {
```

Comment:
@SergeiMikhailovskii `result.isNotEmpty`  would be preferable

Reply:
@anton-novikau result has `QuerySnapshot` type. And there is no `isNotEmpty()` method or field here